### PR TITLE
Automate user signup survey part I

### PR DESCRIPTION
**Pull request by freesteph**
# Description
We went to send signup surveys to:
* 25%
* of our users
* who signed up in the last 24 hours
* who have been active (i.e logged in)

This PR just does that whilst respecting the gateway/usecase pattern in use across the app.

# How to test
* look at the code;
* approve;
* let us get it to staging.

Co-authored with @colinbm .

# Tests
```
/usr/src/app # rspec spec/lib/survey/

Survey::Gateway::Notifications
  when the user signed up via email
    sends an email to them
  when the user signed up via mobile
    sends a text to them

Survey::Gateway::UserDetails
  #fetch
    when the user has been created in the last 24 hours
      but has not logged in
        does not include them
      and has logged in
        includes them
        only returns 25% of users
    when the user was created more than 24 hours ago
      does not include them
    when the user has already been sent the survey
      does not include them
    when the user is sponsored
      does not include them
  #mark_as_sent
    updates the survey_sent_at attribute

Survey::UseCase::SendSurveys
I, [2020-11-13T12:50:27.216814 #270]  INFO -- : [active-users-signup-survey] sending email to 2 users.
  calls the user_details_gateway's fetch method
I, [2020-11-13T12:50:27.218369 #270]  INFO -- : [active-users-signup-survey] sending email to 2 users.
  calls the notifications gateway with each value received
I, [2020-11-13T12:50:27.219089 #270]  INFO -- : [active-users-signup-survey] sending email to 2 users.
  calls the user_details_gateway's mark_as_sent method

Finished in 1.97 seconds (files took 1.67 seconds to load)
12 examples, 0 failures

Coverage report generated for RSpec to /usr/src/app/coverage. 136 / 145 LOC (93.79%) covered.
/usr/src/app # 
```

### CODE REVIEW

File: spec/lib/survey/gateway/notifications_spec.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      phone_number: mobile_user.contact,
+      template_id: "mobile-mock-template",
+    }
+  end
+
+  let(:notify_mobile_stub) do
+    stub_request(:post, notify_mobile_url)
+      .with(body: notify_mobile_request)
+      .to_return(status: 200, body: {}.to_json)
+  end
```
Comment (alan-gds) [inline spec/lib/survey/gateway/notifications_spec.rb:26]: This block (notify_mobile_stub) should probably be moved to the "when the user signed up by mobile" context.

**Review by alan-gds — APPROVED** [review]
Looks good to me.  One minor style quibble, but I'm not too fussed about that.

File: docker-compose.development.yml [hunk 2]
```diff
@@ -6,3 +6,6 @@ services:
     - ".:/usr/src/app"
     ports:
       - "8080:8080"
+  db:
+    ports:
+      - "3306:3306"
```
Comment (sarahseewhy) [inline docker-compose.development.yml:11]: This for my curiosity: why do we need to open the port?

**Review by sarahseewhy — APPROVED** [review]
Left a question but it doesn't block approval.

File: docker-compose.development.yml (hunk 2 above)
Comment (colinbm) [reply docker-compose.development.yml:11]: Ah it doesn't - this has been left over from a WIP branch when I was poking around the database with a GUI client.

**Pull request merged**